metals/src/main/scala/scala/meta/internal/metals/ProjectMetalsLspService.scala

```diff
@@ -436,6 +441,34 @@ class ProjectMetalsLspService(
     }
   }
 
+  private val previousCreateOrModify = TrieMap[AbsolutePath, String]()
+
+  // If code is genereted we might get a lot of `CreateOrModify`
+  // events, even if the code doesn't really change
+  private def didContentChange(event: FileWatcherEvent): Boolean = {
+    val path = AbsolutePath(event.path)
+    event.eventType match {
+      case EventType.CreateOrModify =>
+        try {
+          val md5 = MD5.bytesToHex(FileIO.readAllBytes(path))
```

Comment:
Later in `onChange` the file is read again (if the event isn't ignored). Should I try to deduplicate these reads?